lib/psci/psci_common.c

```diff
@@ -395,6 +395,8 @@ void psci_do_state_coordination(unsigned int end_pwrlvl,
 	plat_local_state_t target_state, *req_states;
 
 	assert(end_pwrlvl <= PLAT_MAX_PWR_LVL);
+	if (end_pwrlvl > PLAT_MAX_PWR_LVL)
+		return;
```

Comment:
The logic for leaving the assert() in is that in the future if someone made a change and there was actually an above array bounds error it would be hard to detect with just a return from a function. An assert() would ensure that the issue was caught (assuming it's compiled in).

If you are happy with using a panic() call instead of a return I'll use that instead and remove the assert().

As for not adding code, I'm not sure of anyway to do that besides disabling the check which I think is a bad idea. I'll re-spin the pull request with a panic() and remove the assert().